python/lsst/daf/butler/script/queryDatasets.py

```diff
@@ -111,26 +104,29 @@ def getAstropyTable(self, datasetTypeName: str) -> AstropyTable:
             None,
             None,
             str,
-            *[typeMap.get(type(value)) for value in refInfo.datasetRef.dataId.full_values],
+            *[typeMap.get(type(value)) for value in ref.dataId.full_values],
         ]
-        if refInfo.uri:
+        if self.datasetRefs[ref]:
             columnNames.append("URI")
             columnTypes.append(None)
 
         rows = []
-        for refInfo in self.datasetRefs:
+        for ref, uri in self.datasetRefs.items():
             row = [
                 datasetTypeName,
-                refInfo.datasetRef.run,
-                str(refInfo.datasetRef.id),
-                *refInfo.datasetRef.dataId.full_values,
+                ref.run,
+                str(ref.id),
+                *ref.dataId.full_values,
             ]
-            if refInfo.uri:
-                row.append(refInfo.uri)
+            if uri:
+                row.append(uri)
             rows.append(row)
 
         dataset_table = AstropyTable(np.array(rows), names=columnNames, dtype=columnTypes)
-        return sortAstropyTable(dataset_table, dimensions, ["type", "run"])
+        if sort:
+            return sortAstropyTable(dataset_table, dimensions, ["type", "run"])
```

Comment:
I had no idea we had a separate sorting code path for `order_by` on the CLI that did it all in memory, and I'm not thrilled by that, in terms of the maintenance load and keeping behavior consistent across the two code paths so we have the freedom to change the implementation behind the scenes (it already looks inconsistent in at least not supporting descending sorts with `-identifier` syntax).  I also have no idea why it's trying to sort by spatial columns (what does that even mean?) and the list of dimensions it takes is not the right way to work out spatial/temporal things (which may be dimension elements, not dimensions).

I do think it might make sense to do some in-memory sorting to make output order deterministic when `order_by` is not given, but that's a lot simpler that reimplementing `order_by` in Python.

Fixing all that in the existing CLI commands is out of scope here, but I think I'd prefer we not add new functionality that depends on it here.  For now, could we support `--order-by` in `query-datasets` only on single-dataset-type queries (which we can do in the DB), and revisit it for multiple-dataset-type queries when we have the new Python methods for that?

Reply:
It's only ever sorting each table by dataset type and not trying to sort across dataset types. Nate P was trying to make it so that the results were returned in a consistent way each time and it didn't seem to be a huge issue given people shouldn't be querying for millions of rows on the command line.  If the user specifies `--order-by` no sorting happens here so I think that's what you are asking for and if someone asks to order by with a dimension that is not part of the dataset type the query breaks before any of this code is hit.